From 87db4f8680222d2579d0ffb5fe507231f88d8aa5 Mon Sep 17 00:00:00 2001
From: PengpengSun <40026211+PengpengSun@users.noreply.github.com>
Date: Sat, 10 Aug 2024 03:32:40 +0800
Subject: [PATCH] Revert "fix(vmware): Set IPv6 to dhcp when there is no IPv6
 addr (#5471)" (#5596)

RH-Author: xiachen <xiachen@redhat.com>
RH-MergeRequest: 108: Revert "fix(vmware): Set IPv6 to dhcp when there is no IPv6 addr (#5471)" (#5596)
RH-Jira: RHEL-54373
RH-Acked-by: Emanuele Giuseppe Esposito <eesposit@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Commit: [1/1] 96cccfca3d29c95d9c491d5995b7aa96adec4621 (xiachen/cloud-init-centos)

This reverts commit 2b6fe6403db769de14f7c7b7e4aa65f5bea8f3e0.

When there is no IPv6 set to dhcp explicitly, NetworkManager keyfile
defaults to method=auto, may-fail=true. When there is Ipv6 set to dhcp
explictily, NetworkManager keyfile will be set to
method=auto, may-fail=false. The default settings are what we want, so
revert the previous change to keep IPv6 not set explicitly.

(cherry picked from commit 65014b97420b41dcb6e7ea17c66bb2539f9b09fc)
Signed-off-by: Amy Chen <xiachen@redhat.com>
---
 .../sources/helpers/vmware/imc/config_nic.py  |  2 +-
 .../sources/vmware/test_vmware_config_file.py | 68 +++++--------------
 2 files changed, 18 insertions(+), 52 deletions(-)

diff --git a/cloudinit/sources/helpers/vmware/imc/config_nic.py b/cloudinit/sources/helpers/vmware/imc/config_nic.py
index 254518af..b07214a2 100644
--- a/cloudinit/sources/helpers/vmware/imc/config_nic.py
+++ b/cloudinit/sources/helpers/vmware/imc/config_nic.py
@@ -207,7 +207,7 @@ class NicConfigurator:
         """
 
         if not nic.staticIpv6:
-            return ([{"type": "dhcp6"}], [])
+            return ([], [])
 
         subnet_list = []
         # Static Ipv6
diff --git a/tests/unittests/sources/vmware/test_vmware_config_file.py b/tests/unittests/sources/vmware/test_vmware_config_file.py
index 25d3b093..b53ea96c 100644
--- a/tests/unittests/sources/vmware/test_vmware_config_file.py
+++ b/tests/unittests/sources/vmware/test_vmware_config_file.py
@@ -240,45 +240,27 @@ class TestVmwareConfigFile(CiTestCase):
             elif cfg.get("name") == nic2.get("name"):
                 nic2.update(cfg)
 
-        # Test NIC1
         self.assertEqual("physical", nic1.get("type"), "type of NIC1")
         self.assertEqual("NIC1", nic1.get("name"), "name of NIC1")
         self.assertEqual(
             "00:50:56:a6:8c:08", nic1.get("mac_address"), "mac address of NIC1"
         )
         subnets = nic1.get("subnets")
-        self.assertEqual(2, len(subnets), "number of subnets for NIC1")
-        subnet_ipv4 = subnets[0]
-        self.assertEqual(
-            "dhcp", subnet_ipv4.get("type"), "Ipv4 DHCP type for NIC1"
-        )
-        self.assertEqual(
-            "auto", subnet_ipv4.get("control"), "NIC1 Control type"
-        )
-        subnet_ipv6 = subnets[1]
-        self.assertEqual(
-            "dhcp6", subnet_ipv6.get("type"), "Ipv6 DHCP type for NIC1"
-        )
+        self.assertEqual(1, len(subnets), "number of subnets for NIC1")
+        subnet = subnets[0]
+        self.assertEqual("dhcp", subnet.get("type"), "DHCP type for NIC1")
+        self.assertEqual("auto", subnet.get("control"), "NIC1 Control type")
 
-        # Test NIC2
         self.assertEqual("physical", nic2.get("type"), "type of NIC2")
         self.assertEqual("NIC2", nic2.get("name"), "name of NIC2")
         self.assertEqual(
             "00:50:56:a6:5a:de", nic2.get("mac_address"), "mac address of NIC2"
         )
         subnets = nic2.get("subnets")
-        self.assertEqual(2, len(subnets), "number of subnets for NIC2")
-        subnet_ipv4 = subnets[0]
-        self.assertEqual(
-            "dhcp", subnet_ipv4.get("type"), "Ipv4 DHCP type for NIC2"
-        )
-        self.assertEqual(
-            "auto", subnet_ipv4.get("control"), "NIC2 Control type"
-        )
-        subnet_ipv6 = subnets[1]
-        self.assertEqual(
-            "dhcp6", subnet_ipv6.get("type"), "Ipv6 DHCP type for NIC2"
-        )
+        self.assertEqual(1, len(subnets), "number of subnets for NIC2")
+        subnet = subnets[0]
+        self.assertEqual("dhcp", subnet.get("type"), "DHCP type for NIC2")
+        self.assertEqual("auto", subnet.get("control"), "NIC2 Control type")
 
     def test_get_nics_list_static(self):
         """Tests if NicConfigurator properly calculates network subnets
@@ -303,7 +285,6 @@ class TestVmwareConfigFile(CiTestCase):
                 elif cfg.get("name") == nic2.get("name"):
                     nic2.update(cfg)
 
-        # Test NIC1
         self.assertEqual("physical", nic1.get("type"), "type of NIC1")
         self.assertEqual("NIC1", nic1.get("name"), "name of NIC1")
         self.assertEqual(
@@ -363,7 +344,6 @@ class TestVmwareConfigFile(CiTestCase):
             else:
                 self.assertEqual(True, False, "invalid gateway %s" % (gateway))
 
-        # Test NIC2
         self.assertEqual("physical", nic2.get("type"), "type of NIC2")
         self.assertEqual("NIC2", nic2.get("name"), "name of NIC2")
         self.assertEqual(
@@ -371,18 +351,16 @@ class TestVmwareConfigFile(CiTestCase):
         )
 
         subnets = nic2.get("subnets")
-        self.assertEqual(2, len(subnets), "Number of subnets for NIC2")
+        self.assertEqual(1, len(subnets), "Number of subnets for NIC2")
 
-        subnet_ipv4 = subnets[0]
-        self.assertEqual("static", subnet_ipv4.get("type"), "Subnet type")
+        subnet = subnets[0]
+        self.assertEqual("static", subnet.get("type"), "Subnet type")
         self.assertEqual(
-            "192.168.6.102", subnet_ipv4.get("address"), "Subnet address"
+            "192.168.6.102", subnet.get("address"), "Subnet address"
         )
         self.assertEqual(
-            "255.255.0.0", subnet_ipv4.get("netmask"), "Subnet netmask"
+            "255.255.0.0", subnet.get("netmask"), "Subnet netmask"
         )
-        subnet_ipv6 = subnets[1]
-        self.assertEqual("dhcp6", subnet_ipv6.get("type"), "Subnet type")
 
     def test_custom_script(self):
         cf = ConfigFile("tests/data/vmware/cust-dhcp-2nic.cfg")
@@ -469,10 +447,7 @@ class TestVmwareNetConfig(CiTestCase):
                             "type": "static",
                             "address": "10.20.87.154",
                             "netmask": "255.255.252.0",
-                        },
-                        {
-                            "type": "dhcp6",
-                        },
+                        }
                     ],
                 }
             ],
@@ -523,10 +498,7 @@ class TestVmwareNetConfig(CiTestCase):
                                     "metric": 10000,
                                 }
                             ],
-                        },
-                        {
-                            "type": "dhcp6",
-                        },
+                        }
                     ],
                 }
             ],
@@ -586,10 +558,7 @@ class TestVmwareNetConfig(CiTestCase):
                                     "metric": 10000,
                                 }
                             ],
-                        },
-                        {
-                            "type": "dhcp6",
-                        },
+                        }
                     ],
                 }
             ],
@@ -634,10 +603,7 @@ class TestVmwareNetConfig(CiTestCase):
                             "address": "10.20.87.154",
                             "netmask": "255.255.252.0",
                             "gateway": "10.20.87.253",
-                        },
-                        {
-                            "type": "dhcp6",
-                        },
+                        }
                     ],
                 }
             ],
-- 
2.39.3

